src/incidents/list/ViewIncidentsTable.tsx

```diff
@@ -22,33 +23,100 @@ function ViewIncidentsTable(props: Props) {
     return <Spinner type="DotLoader" loading />
   }
 
+  // filter data
+  const exportData = [{}]
+  let first = true
+  if (data != null) {
+    data.forEach((elm) => {
+      const entry = {
+        code: elm.code,
+        date: format(new Date(elm.date), 'yyyy-MM-dd hh:mm a'),
+        reportedBy: elm.reportedBy,
+        reportedOn: format(new Date(elm.reportedOn), 'yyyy-MM-dd hh:mm a'),
+        status: elm.status,
+      }
+      if (first) {
+        exportData[0] = entry
+        first = false
+      } else {
+        exportData.push(entry)
+      }
+    })
+  }
+
+  function downloadCSV() {
+    const fields = Object.keys(exportData[0])
+    const opts = { fields }
+    const parser = new Parser(opts)
+    const csv = parser.parse(exportData)
+    console.log(csv)
+
+    const incidentsText = t('incidents.label')
+    const filename = incidentsText.concat('.csv')
+
+    const text = csv
+    const element = document.createElement('a')
+    element.setAttribute('href', `data:text/plain;charset=utf-8,${encodeURIComponent(text)}`)
+    element.setAttribute('download', filename)
+
+    element.style.display = 'none'
+    document.body.appendChild(element)
+
+    element.click()
+
+    document.body.removeChild(element)
```

Comment:
We should move this to some reusable component that adds the `a` to the DOM and click the link when the data becomes available.

Reply:
little confused how to do this as well. Sounds like a good idea but could you elaborate please?

Reply:
It would be something like this: 

```ts
interface Props {
	data: any[],
    fileName: string
}

const DownloadLink = (props: Props) => {
	useEffect(() => {

	}, [data, fileName]);


}

Reply:
It would be something like this: 

```ts
interface Props {
	data: any[],
    fileName: string
}

const DownloadLink = (props: Props) => {
	useEffect(() => {
	    const filename = incidentsText
	      .concat('-')
	      .concat(format(new Date(Date.now()), 'yyyy-MM-dd--hh-mma'))
	      .concat('.csv')

	    const text = csv
	    const element = document.createElement('a')
	    element.setAttribute('href', `data:text/plain;charset=utf-8,${encodeURIComponent(text)}`)
	    element.setAttribute('download', filename)
	
	    element.style.display = 'none'
	    document.body.appendChild(element)
   		element.click()

	    return document.body.removeChild(element)
	}, [data, fileName]);
}

I haven't had a chance to run this, so there might some small issues with it. But this is the general idea. I can provide a working example later.

Reply:
will give a crack at this in a day or two